File: neuron-di-guice/src/test/java/global/tranquillity/neuron/di/guice/it/GreetingModuleTestSuite.java
package global.tranquillity.neuron.di.guice.it;

import com.google.inject.Module;
import global.tranquillity.neuron.di.api.Neuron;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.*;

@Neuron
public interface GreetingModuleTestSuite extends ModuleTestSuite {

    @Override
    default Module module() { return new GreetingModule(); }

    default void test() {
        final Greeting greeting = getInstance(Greeting.class);
        assertThat(getInstance(Greeting.class), is(sameInstance(greeting)));
        assertThat(greeting.formatter(), is(instanceOf(RealFormatter.class)));
        assertThat(greeting.formatter(), is(sameInstance(greeting.formatter())));
        assertThat(greeting.message(), is("Hello Christian!"));
    }
}

File: neuron-di-guice/src/test/scala/global/tranquillity/neuron/di/guice/scala/it/GreetingModuleTestSuite.scala
package global.tranquillity.neuron.di.guice.scala.it

import javax.inject.Singleton

import global.tranquillity.neuron.di.api.Neuron
import global.tranquillity.neuron.di.guice.it.{Formatter, Greeting, RealFormatter, GreetingModuleTestSuite => jGreetingModuleTestSuite}
import global.tranquillity.neuron.di.guice.scala.{NeuronModule, _}

@Neuron
trait GreetingModuleTestSuite extends jGreetingModuleTestSuite {

  def module = new NeuronModule {

    def configure() {
      bindNeuronClass[Greeting].inScope[Singleton]
      bindClass[Formatter].toClass[RealFormatter]
      bindConstantNamed("format").to("Hello %s!")
    }
  }
}
